test/officerManager.js

```diff
@@ -0,0 +1,336 @@
+const User = require('../api/models/User')
+// Require the dev-dependencies
+const chai = require('chai')
+const chaiHttp = require('chai-http')
+const server = require('../server')
+let serverInstance = null
+let app = null
+const expect = chai.expect
+// tools for testing
+const tools = require('../util/testing-utils/tools.js')
+
+chai.should()
+chai.use(chaiHttp)
+
+function initializeServer () {
+  serverInstance = new server.Server()
+  serverInstance.openConnection()
+  app = serverInstance.getServerInstance()
+}
+
+function terminateServer (done) {
+  serverInstance.closeConnection(done)
+}
+
+// Our parent block
+describe('OfficerManager', () => {
+  before(done => {
+    initializeServer(app)
+
+    // Before each test we empty the database
+    tools.emptySchema(OfficerManager)
+    tools.emptySchema(User)
+    done()
+  })
+  after(done => {
+    terminateServer(done)
+  })
+
+  let token = ''
+
+  describe('/POST submit', () => {
+    it('Should register a user', done => {
+      const user = {
+        email: 'test@test.com',
+        password: 'pass',
+        firstName: 'first-name',
+        lastName: 'last-name'
+      }
+
+      chai
+        .request(app)
+        .post('/api/user/register')
+        .send(user)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should log a user in and get a token', done => {
+      const user = {
+        email: 'test@test.com',
+        password: 'pass'
+      }
+      chai
+        .request(app)
+        .post('/api/user/login')
+        .send(user)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+          res.body.should.be.a('object')
+          res.body.should.have.property('token')
+          token = res.body.token
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should not return statusCode 200 when the required fields are not set', done => {
+      const form = {}
+      chai
+        .request(app)
+        .post('/api/officerManager/submit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.not.have.status(200)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return statusCode 200 when all required fields are filled in', done => {
+      const form = {
+        name: 'pinkUnicorn',
+        email: 'test@test.com',
+        level: 2,
+        team: 'dev',
+        major: 'major',
+        token: token
+      }
+      chai
+        .request(app)
+        .post('/api/officerManager/submit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+  })
+
+  describe('/POST GetForm', () => {
```

Comment:
for every API that requires a token, please also test that by submitting a form without a token will return `UNAUTHORIZED`.

Reply:
no token is needed for get request